sdcclient/_monitor.py

```diff
@@ -306,6 +306,46 @@ def get_dashboards(self):
         res = requests.get(self.url + self._dashboards_api_endpoint, headers=self.hdrs, verify=self.ssl_verify)
         return self._request_result(res)
 
+    def get_dashboard(self, dashboard_id):
+        '''**Description**
+            Return a dashboard with the pased in ID. This includes the dashboards created by the user and the ones shared with them by other users.
+
+        **Success Return Value**
+            A dictionary containing the requested dashboard data.
+
+        **Example**
+            `examples/dashboard_basic_crud.py <https://github.com/draios/python-sdc-client/blob/master/examples/dashboard_basic_crud.py>`_
+        '''
+        id_str = dashboard_id if isinstance(dashboard_id, type(str)) else str(dashboard_id)
+        res = requests.get(self.url + self._dashboards_api_endpoint + "/" + id_str, headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+    def update_dashboard(self, dashboard_data, dashboard_id=None):
+        '''**Description**
+            Updates dashboard with provided in data. Dashboard ID is extracted from data if ID is not explicitly provided
+
+        **Success Return Value**
+            A dictionary containing the updated dashboard data.
+
+        **Example**
+            `examples/dashboard_basic_crud.py <https://github.com/draios/python-sdc-client/blob/master/examples/dashboard_basic_crud.py>`_
+        '''
+        if not dashboard_id:
+            if 'id' not in dashboard_data:
+                return [False, "Invalid dashboard format. ID must be present as paramter or in data set."]
+            dashboard_id = dashboard['id'] if isinstance(dashboard_data['id'], type(str)) else str(dashboard_data['id'])
+        elif not isinstance(dashboard_id, type(str)):
+            dashboard_id = str(dashboard_id)
+
+        dashboard_data_clone = copy.deepcopy(dashboard_data)
+        if 'id' in dashboard_data_clone:
+            del dashboard_data_clone['id']
+        if 'version' in dashboard_data_clone:
+            del dashboard_data_clone['version']
+
+        res = requests.put(self.url + self._dashboards_api_endpoint + "/" + dashboard_id, headers=self.hdrs, verify=self.ssl_verify, data=json.dumps({'dashboard': dashboard_data_clone}))
```

Comment:
The `update` method should be required to work on a valid dashboard. As such, the dashboard configuration must contain everything needed for the update.

The method could be safely simplified to:

```
def update_dashboard(self, dashboard):
        res = requests.put(self.url + self._dashboards_api_endpoint + "/" + str(dashboard['id']), headers=self.hdrs, verify=self.ssl_verify, data=json.dumps({'dashboard': dashboard}))```

Also, note that the update would fail otherwise because:
1. the API requires the payload to contain the `id` property which must match the `id` parameter in the URL, and
2. the API requires the payload to contain the `version` property to avoid conflicting updates of the same dashboard